### PR TITLE
Adding testing with custom-build netcdf and HDF5 versions.

### PR DESCRIPTION
**Pull request by edwardhartnett**
Part of #467 

Adding testing with custom-build netcdf and HDF5 versions.

### CODE REVIEW

**Comment by kgerheiser** [comment]
This will be helpful for the Intel build

**Comment by edwardhartnett** [comment]
@kgerheiser this works now.

Should it be left as a separate workflow? Or do you want to try merging it into the existing workflow?

(There are other versions of netcdf we may check against, as well as perhaps multiple versions of ESMF we might want to test against.)

File: .github/workflows/netcdf-4.7.4.yml [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+        python3 -m pip install gcovr
+
+    - name: cache-esmf
+      id: cache-esmf
+      uses: actions/cache@v2
+      with:
+        path: ~/esmf
+        key: esmf-${{ runner.os }}-8.0.1-2
+
+    - name: build-esmf
```
Comment (kgerheiser) [inline .github/workflows/netcdf-4.7.4.yml:68]: Build NetCDF should come before building ESMF. If ESMF weren't cached it would fail building.

**Review by kgerheiser — CHANGES_REQUESTED** [review]

File: .github/workflows/netcdf-4.7.4.yml (hunk 1 above)
Comment (edwardhartnett) [reply .github/workflows/netcdf-4.7.4.yml]: Good catch! I have fixed, and pointed ESMF at the installed netCDF build.

**Comment by edwardhartnett** [comment]
@kgerheiser please approve and we can merge.

**Comment by kgerheiser** [comment]
What do you think of putting this into the existing workflow instead of creating a new one? I am currently testing it on a branch.

**Comment by edwardhartnett** [comment]
Fine if we can.

But I would like to add testing for netcdf-c-4.8.0 as well. ;-) Can you do that at the same time?

It builds with the same version of Fortran as 4.7.4.

**Comment by kgerheiser** [comment]
Sure, adding NetCDF 4.8.0 is easy. I'll add it as a matrix.

**Comment by kgerheiser** [comment]
See #503

**Pull request closed without merging**